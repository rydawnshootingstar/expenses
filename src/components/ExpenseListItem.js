import React from 'react'
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';
import {Link} from 'react-router-dom';


const ExpenseListItem = ({id, description, amount, createdAt})=> (
    <div>
    <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>Cost: {amount}</p>
        <p>Created At: {createdAt}</p>
    </div>
);


export default (ExpenseListItem);