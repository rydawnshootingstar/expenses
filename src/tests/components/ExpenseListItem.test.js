import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ExpenseListItem from '../../components/ExpenseListItem'
import {testExpenses}from '../fixtures';

test('should render an expense with info',()=> {
    const wrapper = shallow(<ExpenseListItem {...testExpenses[0]}/>);
    console.log(toJSON(wrapper));
    expect(toJSON(wrapper)).toMatchSnapshot();
});