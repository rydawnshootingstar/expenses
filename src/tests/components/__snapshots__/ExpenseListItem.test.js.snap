// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense with info 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      exp 1
    </h3>
  </Link>
  <p>
    Cost: 
    111
  </p>
  <p>
    Created At: 
    0
  </p>
</div>
`;
